service/resource/cloudformationv2/create.go

```diff
@@ -0,0 +1,90 @@
+package cloudformationv2
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/keyv2"
+)
+
+func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+	cluster, err := keyv2.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	stackInput, err := toCreateStackInput(createChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	_, err = r.Clients.CloudFormation.CreateStack(&stackInput)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	err = r.HostClients.CloudFormation.WaitUntilStackCreateComplete(&cloudformation.DescribeStacksInput{
+		StackName: stackInput.StackName,
+	})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "debug", "creating AWS cloudformation stack: created")
+
+	// Create host post-main stack. It includes the peering routes, which need resources from the
+	// guest stack to be in place before it can be created.
+	err = r.createHostPostStack(cluster)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := keyv2.ToCustomObject(obj)
+	if err != nil {
+		return cloudformation.CreateStackInput{}, microerror.Mask(err)
+	}
+
+	desiredStackState, err := toStackState(desiredState)
+	if err != nil {
+		return cloudformation.CreateStackInput{}, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "debug", "finding out if the main stack should be created")
+
+	createState := cloudformation.CreateStackInput{
+		StackName: aws.String(""),
```

Comment:
Why is the stack name empty here?

Reply:
Good catch! not needed, removed here and in update too